activerecord/lib/active_record/relation/batches.rb

```diff
@@ -274,46 +304,55 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
           end
         end
 
-        batch_relation = batch_condition(
-          relation, primary_key, primary_key_offset, order == :desc ? :lt : :gt
-        )
+        batch_orders_copy = batch_orders.dup
```

Comment:
I was wondering if this is a precaution or we must `dup` the array as somewhere else we expect `batch_orders` to keep original size. 
I'm not concerned about it as I still find `.pop` solution being more readable compared to `current_index == last_index` so I'm okay with keeping `.dup` 

But I wouldn't be against getting back to index comparison approach if someone has strong opinion against `.dup`